api/modules/budgeting.js

```diff
@@ -0,0 +1,113 @@
+const db = require('../models')
+const moment = require('moment')
+const clientManagement = require('./clientManagement')
+
+const budgeting = module.exports = (() => {
+
+    const createPayment = async ({ paymentInformation }) => {
+        const client = await clientManagement.getClientWithExternalId({ id: paymentInformation.customer_id })
+
+        const {
+            amount,
+            external_uuid,
+            date_incurred,
+            date_paid,
+            external_uuid_type
+        } = paymentInformation
+
+        if (client) {
+            return db.models.Payment.create({
+                amount,
+                external_uuid,
+                date_incurred: moment(date_incurred['_d']).format('YYYY-MM-DD HH:mm:ss'),
+                date_paid: date_paid ? moment(date_paid['_d']) : null,
+                client_id: client.id,
+                external_uuid_type
+            })
+        }
+    }
+
+    const deletePaymentByStripeInvoiceId = (params) => {
+        let deletedPayment
+        try {
+            deletedPayment = db.models.Payment.destroy({
+                where: {
+                    external_uuid: params.invoiceId
+                }
+            })
+        } catch {
+            throw 'Failed deleting from database'
+        }
+        return deletedPayment
+    }
+
+    const getPaymentWithExternalId = (params) => {
+        return db.models.Payment.findOne({
+            where: {
+                external_uuid: params.id
+            },
+            raw: true,
+        })
+    }
+
+    const getPaymentWithId = (params) => {
+        return db.models.Payment.findByPk(params.id, {
+            raw: true
+        })
+    }
+
+    const updateDatePaidPayment = async ({ paymentObjectPayload }) => {
+        const paymentInformation = {
+            date_paid: paymentObjectPayload.webhooks_delivered_at,
+            external_uuid: paymentObjectPayload.id
+        }
+        const paymentToUpdate = {}
+        if (paymentInformation.external_uuid) {
+            Object.assign(paymentToUpdate, await getPaymentWithExternalId({ id: paymentInformation.external_uuid }))
+        } else {
+            Object.assign(paymentToUpdate, await getPaymentWithId({ id: paymentInformation.id }))
+        }
+
+        paymentToUpdate.date_paid = paymentInformation.date_paid
+        await db.models.Payment.update({
+            date_paid: moment(paymentToUpdate.date_paid, 'YYYY-MM-DD')
+        }, {
+            where: {
+                id: paymentToUpdate.id
+            }
+        })
+    }
+
+    const updatePaymentByStripeInvoiceId = async ({ paymentObjectPayload }) => {
+        const datePaidOverride = paymentObjectPayload.custom_fields[findIndex(paymentObjectPayload.custom_fields, { 'name': 'date_paid' })]
```

Comment:
@mecoccaro As a recommendation, I would try to make the lines of code more concise to avoid spaghetti code, you could either break this code into multiple lines 
```
const datePaidOverride = paymentObjectPayload.custom_fields[
   findIndex(
      paymentObjectPayload.custom_fields, 
      { 'name': 'date_paid' }
   )
]
```
or create a constant with a significant name and store the value of `findIndex(paymentObjectPayload.custom_fields, { 'name': 'date_paid' })` in it and then make the reference with that value

Reply:
Also `'name'` as a property name we should not use single quotes around it so we don't confuse it for a string. so it'd be just `{ name: 'date_paid' }`